packages/remix/src/utils.ts

```diff
@@ -18,6 +34,60 @@ export function findConfig(dir: string, basename: string): string | undefined {
   return undefined;
 }
 
+function isEdgeRuntime(runtime: string): boolean {
+  return runtime === 'edge' || runtime === 'experimental-edge';
+}
+
+export function getResolvedRouteConfig(
+  route: ConfigRoute,
+  routes: RouteManifest,
+  configs: Map<ConfigRoute, BaseFunctionConfig | null>
+): ResolvedRouteConfig {
+  let runtime: ResolvedRouteConfig['runtime'] | undefined;
+  let regions: ResolvedRouteConfig['regions'];
+  let maxDuration: ResolvedNodeRouteConfig['maxDuration'];
+  let memory: ResolvedNodeRouteConfig['memory'];
+
+  for (const currentRoute of getRouteIterator(route, routes)) {
+    const staticConfig = configs.get(currentRoute);
+    if (staticConfig) {
+      if (typeof runtime === 'undefined' && staticConfig.runtime) {
+        runtime = isEdgeRuntime(staticConfig.runtime) ? 'edge' : 'nodejs';
+      }
```

Comment:
**question (non-blocking):** where should we validate that `maxDuration` and `memory` are only used when `runtime === 'edge'`?

We could do it here, but it probably makes more sense in `static-config`'s `getConfig` function.

Reply:
Currently those values would just be ignored. We could improve upon that in the future if needed though.

Reply:
Right. Someone could configure them and wonder why they aren't doing anything. It's a UX bug, in my opinion.